src/biogeophys/BalanceCheckMod.F90

```diff
@@ -272,7 +283,10 @@ subroutine BalanceCheck( bounds, &
                   - (forc_rain_col(c)        &
                   + forc_snow_col(c)         &
                   + qflx_floodc(c)           &
-                  + qflx_irrig(c)            &
+                  + qflx_irrig_drip(c)       &
```

Comment:
One more comment: Looking back through the changes in BalanceCheckMod, I realized that these changes are a little incongruous with the surrounding code: For the most part, the only water fluxes we consider here are ones that come from outside CTSM (rain, snow, flood) or leave CTSM (runoff, evaporation, etc.). But some of these irrigation fluxes can just be due to rearrangements of water within the system (extracting from groundwater and adding to the top of the soil, for example).

I think you should get rid of all of these qflx_irrig terms and instead just add qflx_sfc_irrig. I think this should amount to the same thing, since I think qflx_sfc_irrig = (qflx_irrig_drip + qflx_irrig_sprinkler - qflx_gw_uncon_irrig - qflx_gw_con_irrig). However, this is more congruous with the surrounding code. And, more importantly, this will do a better job at catching balance errors. (For example, I think that, as currently written, you could put something like 'qflx_irrig_drip = qflx_irrig_drip + 1' at the end of ApplyIrrigation, and this balance check as written would still pass, when it shouldn't.)